lib/Doctrine/Common/Proxy/ProxyGenerator.php

```diff
@@ -1116,6 +1116,33 @@ private function getParameterType(ReflectionParameter $parameter)
         return $this->formatType($parameter->getType(), $declaringFunction, $parameter);
     }
 
+    /**
+     * @return string
+     */
+    private function getParameterDefaultValue(ReflectionParameter $parameter)
+    {
+        if (! $parameter->isDefaultValueAvailable()) {
+            return '';
+        }
+
+        if (PHP_VERSION_ID < 80100) {
+            return ' = ' . var_export($parameter->getDefaultValue(), true);
+        }
+
+        $value = rtrim(substr(explode('$' . $parameter->getName() . ' = ', (string) $parameter, 2)[1], 0, -2));
+        $parts = preg_split('{(\'(?:[^\'\\\\]*(?:\\\\.[^\'\\\\]*)*)\')}', $value, -1, PREG_SPLIT_DELIM_CAPTURE);
```

Comment:
It would be nice to use the `x` delimiter and add comments inside that regex. If that makes it become multiline, then maybe also use nowdoc? That way there won't be a need for escaping the single quotes.

Reply:
Same thing as above